test/exercise/fp/solution.rb

```diff
@@ -4,12 +4,18 @@ class << self
       # Обратиться к параметрам фильма можно так:
       # film["name"], film["rating_kinopoisk"], film["rating_imdb"],
       # film["genres"], film["year"], film["access_level"], film["country"]
-      def rating(_array)
-        0
+      def rating(array)
+        films_sort = array.select do |film|
+          film['rating_kinopoisk'].to_f.positive? &&
+            film['country'].to_s.split(',').length > 1
+        end
+        result = films_sort.map { |film| film['rating_kinopoisk'].to_f }.reduce(:+)
+        result / films_sort.length
       end
 
-      def chars_count(_films, _threshold)
-        0
+      def chars_count(films, threshold)
+        films.select { |film| film['rating_kinopoisk'].to_f >= threshold }
+             .map { |film| film['name'] }.join('').count('и')
```

Comment:
complexity is 4*N, it's quite high. but it **works**

FYI: there is an algorithm with N iterations for it